Automattic/jetpack
# Fix VaultPress query refresh

#### Changes proposed in this Pull Request:

* Fixed a bug that caused endless fetching of VaultPress data in some circumstances, causing cards on the jetpack dashboard to flicker

https://user-images.githubusercontent.com/937354/166066158-b3cc10c5-10ac-40b3-8b70-cb689426794a.mp4

* I have added a root cause analysis of the bug later in the comments in this PR: https://github.com/Automattic/jetpack/pull/24194#issuecomment-1113840687


#### Jetpack product discussion


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Have a Jetpack site without this PR
* Install and activate the VaultPress plugin, but don't register
* Visit the JP Dashboard (`/wp-admin/admin.php?page=jetpack#/dashboard`)
* Note that some cards are flickering
* Repeat steps with this PR; this time, the flickering should not be present
* VaultPress related functionality in Jetpack should continue to work
  * ^ The above is vague because I know very little about it.
